### PR TITLE
Delegate properties to atomics 

### PR DESCRIPTION
**Pull request by mvicsokolova**

### CODE REVIEW

File: kotlinx-coroutines-core/api/kotlinx-coroutines-core.api [hunk 1]
```diff
@@ -62,6 +62,7 @@ public class kotlinx/coroutines/CancellableContinuationImpl : kotlin/coroutines/
 	public fun getContinuationCancellationCause (Lkotlinx/coroutines/Job;)Ljava/lang/Throwable;
 	public final fun getResult ()Ljava/lang/Object;
 	public fun getStackTraceElement ()Ljava/lang/StackTraceElement;
+	public final fun getState$kotlinx_coroutines_core ()Ljava/lang/Object;
```
Comment (qwwdfsad) [inline kotlinx-coroutines-core/api/kotlinx-coroutines-core.api:65]: Please investigate why these getters are now considered public. It seems like a specific attribute is missing from metadata

**Comment by qwwdfsad** [comment]
Any update about this one?

**Comment by mvicsokolova** [comment]
Depends on this: #3501
Waiting for the declarations visibility fix in `1.7.21`

**Comment by mvicsokolova** [comment]
There was a bug revealed on JS IR side with Kotlin `1.7.20`: private delegated properties caused an error at runtime, the corresponding issue: https://github.com/Kotlin/kotlinx-atomicfu/issues/260.

The fix in the compiler plugin will be provided with Kotlin `1.8.0-RC`

**Comment by qwwdfsad** [comment]
jvm-ir transformer along with 1.8.10 Kotlin are in master

**Comment by qwwdfsad** [comment]
Can be opened as a separate PR

**Pull request closed without merging**